src/OpenLoco/src/Audio/Audio.cpp

```diff
@@ -1058,28 +1064,29 @@ namespace OpenLoco::Audio
                 playlist.push_back(i);
             }
         }
+
+        return playlist;
     }
 
     std::vector<uint8_t> makeSelectedPlaylist()
     {
         using MusicPlaylistType = Config::MusicPlaylistType;
 
-        std::vector<uint8_t> playlist = std::vector<uint8_t>();
-
         switch (Config::get().old.musicPlaylist)
         {
             case MusicPlaylistType::currentEra:
-                addCurrentEraSongsToPlaylist(playlist);
+            default:
```

Comment:
What exactly is default for? With the strong enum we shouldn't ever run into arbitrary values unless there is a case missing, in that case it should be explicit with fall through annotation.

Reply:
The compiler doesn't like it when it's technically possible for the function to not return any value.

I thought about having the default be to throw an exception, however I realised that this might cause players with an invalid config from fixing their setting, as the game might crash before they get to the options screen to change it.

Returning an empty vector is also an option, though defaulting to the 'current era' tracks appears to be the original Locomotion's default by my testing (via editing GAME.CFG file with a hex editor).

Reply:
There should be no invalid option, if its possible to do that then we have to fix the config import code to keep those things sanitized to only valid choices. The compiler shouldn't complain when it covers all the cases.

Reply:
From https://developercommunity.visualstudio.com/t/erroneous-c4715-warning-returning-from-switch-case/1167679, it sounds like the compiler issue I'm running into is "works as designed"

Reply:
Is throwing a Exception::RuntimeError preferable?

Reply:
Yeah you can place that at the bottom, don't add a default as that will silence otherwise the warning for a missing enumeration in case we ever add more.